test/openssl/test_pkey_dh.rb

```diff
@@ -89,6 +89,24 @@ def test_dup
     assert_equal dh2.g, dh.g
   end
 
+  def test_match?
+    key1 = Fixtures.pkey("dh1024")
+    key2 = Fixtures.pkey("dh1024")
+
+    key3 = Fixtures.pkey("dh-1")
+
+    assert_equal(true, key1.match?(key2))
+    assert_equal(true, key1.public_key.match?(key2))
+    assert_equal(true, key2.match?(key1))
+    assert_equal(true, key2.public_key.match?(key1))
+
+    assert_equal(false, key1.match?(key3))
+
+    assert_raise(TypeError) do
+      key1.match?("DH Key")
+    end
+  end
+
```

Comment:
I think a single test case in test/openssl/test_pkey.rb would suffice since we don't have any algorithm-specific code in C.

Reply:
Agree. Wanted to leave as-is to showcase the EC public_key differs, but wasn't aware you already knew this. Will update.